src/kibana/components/persisted_state/persisted_state.js

```diff
@@ -0,0 +1,105 @@
+define(function (require) {
+  var _ = require('lodash');
+  var errors = require('errors');
+
+  return function () {
+    function validateParent(parent, path) {
+      if (_.isUndefined(parent)) return;
+
+      if (path.length <= 0) {
```

Comment:
Do you want `if (!path.length)` here? If you pass a number or something else stupid as path this won't error